bundler/lib/bundler/lockfile_generator.rb

```diff
@@ -69,10 +69,12 @@ def add_dependencies
     def add_checksums
       out << "\nCHECKSUMS\n"
 
-      empty_store = Checksum::Store.new
-
       definition.resolve.sort_by(&:full_name).each do |spec|
-        out << (spec.source.checksum_store || empty_store)[spec].to_lock
+        next unless spec.source.checksum_store
+        next if (checksums = spec.source.checksum_store[spec.full_name]).empty?
```

Comment:
This is different than the existing specs. Probably causing some of the failures. If we still want gem names with empty checksums then we can move this guard down to only skipping the checksum part.